assets/deepen_shallow_clone_until_ref_is_found_then_check_out

```diff
@@ -24,27 +27,28 @@ readonly tagflag="$3"
 git_dir="$(git rev-parse --git-dir)"
 readonly git_dir
 
-while ! git checkout -q "$ref" &>/dev/null; do
+while ! git checkout -q "$ref~$depth_from_ref" &>/dev/null; do
```

Comment:
Why do we start from `$ref~$depth_from_ref` instead of `$ref` here? Thx.

Reply:
The idea is described in the comment block L15-25. Basically by specifying the depth parameter the user says: give me a shallow repo at ref `x` where both `git checkout x` and `git checkout x~depth_from_ref` will work. Now the problem is that the cloned repo is actually cloned at ref `y` which may be any number of commits behind `x`. It's redundant to deepen until `git checkout x` works, if we deepen until `git checkout x~depth_from_ref` works it's guaranteed that `git checkout x` works too.